File: 047.LeetCode@128_Longest_Consecutive_Sequence/047.LeetCode@128_Longest_Consecutive_Sequence.py
class Solution:
    def longestConsecutive(self, nums: list[int]) -> int:
        if len(nums)==0:return 0
        nums.sort()
        length = maxlength =1
        for i in range (1,len(nums)):
            if nums[i] == nums[i-1]+1:
                length+=1
                maxlength = max(maxlength,length)
            elif nums[i] == nums[i-1]:
                continue
            else:
                length = 1
        return maxlength

File: 047.LeetCode@128_Longest_Consecutive_Sequence/047.LeetCode@128_Longest_Consecutive_Sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums == null || nums.length == 0) return 0;
        Arrays.sort(nums);
        int length = 1, maxlength = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == nums[i - 1]) continue;
            else if (nums[i] == nums[i - 1] + 1) {
                length++;
                maxlength = Math.max(length, maxlength);
            } else {
                length = 1;
            }
        }
        return maxlength;
    }
}